File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/search/viewmodel/SearchViewModelFactory.java
package com.nhatminh.example.architecture.demoarchitecture.search.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.nhatminh.example.architecture.demoarchitecture.repository.DataRepository;

public class SearchViewModelFactory implements ViewModelProvider.Factory {

    private DataRepository repository;

    public SearchViewModelFactory(DataRepository repository) {
        this.repository = repository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new SearchViewModel(repository);
    }
}

File: app/src/main/java/com/nhatminh/example/architecture/demoarchitecture/search/viewmodel/SearchViewModelFactory.kt
package com.nhatminh.example.architecture.demoarchitecture.search.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.nhatminh.example.architecture.demoarchitecture.repository.DataRepository

class SearchViewModelFactory (val repository: DataRepository) : ViewModelProvider.Factory {


    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(SearchViewModel::class.java)) {
            return SearchViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}